pkg/layeredsets/string.go

```diff
@@ -0,0 +1,170 @@
+/*
+Copyright 2020 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package layeredsets
+
+import (
+	"math/rand"
+	"sort"
+
+	"k8s.io/apimachinery/pkg/util/sets"
+)
+
+// String is a layered set implemented as a map of sets.String.
+type String map[int]sets.String
```

Comment:
Sorry for bringing this up quite late, but given how we work with layers, I wonder if this should be a slice of sets instead of a map - the indices are always a growing integer sequence, so using a map seems like a waste for such data structure.

I don't insist on this.

Reply:
thanks for the suggestion, done